USER_GUIDE.md

```diff
@@ -1,7 +1,166 @@
+| `appengineDeployDispatch` | Deploy dispatch configuration. |
+| `appengineDeployDos`      | Deploy dos configuration. |
+| `appengineDeployIndex`    | Deploy datastore index configuration. |
+| `appengineDeployQueue`    | Deploy queue configuration. |
+
+#### Other
+
+| Task                         | Description |
+| ---------------------------- | ----------- |
+| `appengineShowConfiguration` | Print out the appengine standard gradle plugin configuration |
+
+### Configuration
+Once you've configured `gcloud` in the Cloud SDK, no gradle configuration should be needed to run
+and deploy an application, however if you chose to customize your configuration, the standard plugin
+can be configured using the `appengine` configuration closure. 
+
+To print the **default** configuration values, run `appengineShowConfiguration`. It is recommended
+you check the default before setting anything manually.
+
+```groovy
+appengine {
+  tools {
+    // configure the Cloud Sdk tooling
+  }
+  run {
+    // configure local run
+  }
+  stage {
+    // configure staging for deployment
+  }
+  deploy {
+    // configure deployment
+  }
+}
+```
+
+##### Tools
+The `tools` configuration has the following parameters.
+
+| Parameter             | Description |
+| -------------- | ----------- |
+| `cloudSdkHome` | Location of to the cloud sdk, the plugin will try to find a CloudSdkHome is none is specified. |
+
+##### Run
+The `run` configuration has the following parameters. Note that only a subset are valid for Dev App Server version "1"
+and all are valid for Dev App Server version "2-alpha"
+
+Valid for versions "1" and "2-alpha"
+
+| Parameter             | Description |
+| --------------------- | ----------- |
+| ~~`appYamls`~~        | Deprecated in favor of `services` |
+| `jvmFlags`            | JVM flags to pass to the App Server Java process. |
+| `host`                | Application host address. |
+| `port`                | Application host port. |
+| `startSuccessTimeout` | Amount of time in seconds to wait for the Dev App Server to start in the background. |
+| `serverVersion`       | Server versions to use, options are "1" or "2-alpha" |
+| `services`            | List of services to run |
+ 
+Only valid for versions "2-alpha"
+
+| Parameter (2-alpha only) |
+| ------------------------ |
+| `adminHost`              |
+| `adminPort`              |
+| `allowSkippedFiles`      |
+| `apiPort`                |
+| `authDomain`             |
+| `automaticRestart`       |
+| `clearDatastore`         |
+| `customEntrypoint`       |
+| `datastorePath`          |
+| `defaultGcsBucketName`   |
+| `devAppserverLogLevel`   |
+| `logLevel`               |
+| `maxModuleInstances`     |
+| `pythonStartupArgs`      |
+| `pythonStartupScript`    |
+| `runtime`                |
+| `skipSdkUpdateCheck`     |
+| `storagePath`            |
+| `threadsafeOverride`     |
+| `useMtimeFileWatcher`    |
+
+##### Stage
+The `stage` configuration has the following parameters.
```

Comment:
There are flex only staging parameters as well, we could add a similar sentence to what we have for deploy.